### DISCUSSION TITLE
ValueError: The Decimal type doesn't accept collections as value

### DISCUSSION THREAD

**Discussion by nnazumud** [community/community#24863] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I am trying to invoke WSDL to update tracking number for a shipment line in oracle fusion cloud : {<a href="http://xmlns.oracle.com/apps/scm/shipping/shipConfirm/deliveries/shipmentLineService/%7DShipmentLineService" rel="nofollow noopener">http://xmlns.oracle.com/apps/scm/shipping/shipConfirm/deliveries/shipmentLineService/}ShipmentLineService</a> wsdl via python using zeep. Below is the payload that successfully executes in SOAP UI and returns a response. When I invoke the wsdl via python it throws ValueError: The Decimal type doesn’t accept collections as value. Please advice.</p>
<p>SOAP Payload:</p>
<p>&lt;soapenv:Envelope xmlns:soapenv=“<a href="http://schemas.xmlsoap.org/soap/envelope/" rel="nofollow noopener">http://schemas.xmlsoap.org/soap/envelope/</a>” xmlns:typ=“<a href="http://xmlns.oracle.com/apps/scm/shipping/shipConfirm/deliveries/shipmentLineService/types/" rel="nofollow noopener">http://xmlns.oracle.com/apps/scm/shipping/shipConfirm/deliveries/shipmentLineService/types/</a>” xmlns:ship=“<a href="http://xmlns.oracle.com/apps/scm/shipping/shipConfirm/deliveries/shipmentLineService/" rel="nofollow noopener">http://xmlns.oracle.com/apps/scm/shipping/shipConfirm/deliveries/shipmentLineService/</a>”&gt;<br>
&lt;soapenv:Header/&gt;<br>
&lt;soapenv:Body&gt;<br>
&lt;typ:UpdateShipmentLines&gt;<br>
&lt;typ:apiVersionNumber&gt;1.0&lt;/typ:apiVersionNumber&gt;<br>
&lt;typ:InitMsgList&gt;T&lt;/typ:InitMsgList&gt;</p>
<p>&lt;typ:ShipmentLineInformation&gt;<br>
&lt;ship:ShipmentLine&gt;619092&lt;/ship:ShipmentLine&gt;<br>
&lt;ship:TrackingNumber&gt;67481420011234&lt;/ship:TrackingNumber&gt;<br>
&lt;/typ:ShipmentLineInformation&gt;<br>
&lt;/typ:UpdateShipmentLines&gt;<br>
&lt;/soapenv:Body&gt;<br>
&lt;/soapenv:Envelope&gt;</p>
<p>Python Code:</p>
<p>import os<br>
import pprint<br>
import requests<br>
import time<br>
import zeep<br>
import sys<br>
import csv</p>
<p>client = None<br>
username = ‘xxxx’<br>
password = ‘abcd’<br>
wsdl = ‘https://&lt;host&gt;/fscmService/ShipmentLineService?WSDL’<br>
filename = sys.argv[1] <span class="hashtag">#This</span> is the input file with 2 columns, 1 for shipmentline and 1 for the tracking number</p>
<p>def get_client_session(username, password):<br>
session = requests.Session()<br>
session.auth = requests.auth.HTTPBasicAuth(username, password)<br>
transport = zeep.transports.Transport(session=session, timeout=30)<br>
client = zeep.Client(wsdl=wsdl, transport=transport)</p>
<p>return client</p>
<p>with open(filename) as csv_file:</p>
<p>parsed_csv = csv.reader(csv_file, delimiter=’\t’)</p>
<p>for row in parsed_csv:<br>
if (row[0] == “ShipmentLineId”):<br>
print(“Skipped header row”)<br>
print("=" * 100)<br>
else:<br>
shipline = row[0]<br>
trknum = row[1]<br>
shiplineinfo = {<br>
‘ShipmentLine’:shipline,<br>
‘TrackingNumber’:trknum<br>
}</p>
<p>trackinfo = {<br>
‘apiVersionNumber’: 1.0,<br>
‘InitMsgList’: ‘T’,<br>
‘ShipmentLineInformation’: shiplineinfo<br>
}</p>
<p>print(trackinfo)<br>
client = get_client_session(username, password)<br>
try:<br>
response = client.service.UpdateShipmentLines(trackinfo)<br>
print(“Successful update on ShipmentLine #” + row[0] + " track #" + row[1])<br>
print(“Here’s the response:”)<br>
print(response)<br>
print("=" * 100)<br>
except zeep.exceptions.Fault as error:<br>
print(“Failed update on ShipmentLine #” + row[0] + " track #" + row[1])<br>
print(error)<br>
print("*" * 100)</p>
<p>Error Msg:</p>
<h1>$ python updateTrackingNumber.py ShipInfo.tsv<br>
Skipped header row</h1>
<p>{‘ShipmentLineInformation’: {‘TrackingNumber’: ‘850683456’, ‘ShipmentLine’: ‘585068’}, ‘InitMsgList’: ‘T’, ‘apiVersionNumber’: 1.0}<br>
Traceback (most recent call last):<br>
File “updateTrackingNumber.py”, line 49, in &lt;module&gt;<br>
response = client.service.UpdateShipmentLines(trackinfo)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/proxy.py”, line 42, in __call__<br>
self._op_name, args, kwargs)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/wsdl/bindings/soap.py”, line 121, in send<br>
options=options)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/wsdl/bindings/soap.py”, line 68, in _create<br>
serialized = operation_obj.create(*args, **kwargs)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/wsdl/definitions.py”, line 200, in create<br>
return self.input.serialize(*args, **kwargs)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/wsdl/messages/soap.py”, line 78, in serialize<br>
self.body.render(body, body_value)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/elements/element.py”, line 203, in render<br>
self._render_value_item(parent, value, render_path)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/elements/element.py”, line 227, in _render_value_item<br>
return self.type.render(node, value, None, render_path)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/types/complex.py”, line 253, in render<br>
element.render(parent, element_value, child_path)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/elements/indicators.py”, line 241, in render<br>
element.render(parent, element_value, child_path)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/elements/element.py”, line 203, in render<br>
self._render_value_item(parent, value, render_path)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/elements/element.py”, line 227, in _render_value_item<br>
return self.type.render(node, value, None, render_path)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/types/simple.py”, line 74, in render<br>
parent.text = self.xmlvalue(value)<br>
File “/home/system/.local/lib/python2.7/site-packages/zeep/xsd/types/builtins.py”, line 30, in _wrapper<br>
self.__class__.__name__))<br>
ValueError: The Decimal type doesn’t accept collections as value</p>

**Comment by Munyola** [comment]
<p>Thanks for your improvement,but that questions are a bit confused. (1)<a href="http://Oracle.com" rel="nofollow noopener">Oracle.com</a>(2)w3.com- this are two different text protocols.It could be better to concentrate on neighboring site whether they support the issuing on developers retrieve. Github associated sites have got retrieve gateway that will directly launch you to the position of that content,this action are found in any new website page. It will be difficult to handle non associated content that is from independent sites.so if developer is not included among the share option, may they have away of repairing their projects. Otherwise thanks for your struggling keep it up👍<br>
Thanks…</p>

**Comment by nnazumud** [comment, marked as answer]
<p>This issue can be closed. Once I passed the parameters as a named value pair within the service method, the error went away and the service executed successfully.</p>
<p>response = client.service.UpdateShipmentLines(apiVersionNumber=1.0,InitMsgList=‘T’,ShipmentLineInformation=shiplineinfo)</p>
<p>This solved the problem.</p>
<p>Thanks!</p>
